central/complianceoperator/v2/report/manager/complianceReportgenerator/report_gen_impl.go

```diff
@@ -96,15 +101,33 @@ type ResultEmail struct {
 	Clusters   int
 }
 
-func (rg *complianceReportGeneratorImpl) ProcessReportRequest(req *ComplianceReportRequest) {
+func (rg *complianceReportGeneratorImpl) ProcessReportRequest(req *ComplianceReportRequest) error {
 
 	log.Infof("Processing report request %s", req)
+
+	snapshot, found, err := rg.snapshotDS.GetSnapshot(req.Ctx, req.SnapshotID)
+	if err != nil {
+		return errors.Wrap(err, "unable to retrieve the snapshot from the store")
+	}
+	if !found {
+		return errors.New("unable to find snapshot in the store")
+	}
+
 	data := rg.getDataforReport(req)
 
 	zipData, err := format(data.ResultCSVs)
 	if err != nil {
-		log.Errorf("Error zipping compliance reports for scan config %s:%s", req.ScanConfigName, err)
-		return
+		snapshot.GetReportStatus().RunState = storage.ComplianceOperatorReportStatus_FAILURE
+		snapshot.GetReportStatus().ErrorMsg = reportUtils.ErrReportGeneration.Error()
+		snapshot.GetReportStatus().CompletedAt = protocompat.TimestampNow()
+		if dbErr := rg.snapshotDS.UpsertSnapshot(req.Ctx, snapshot); dbErr != nil {
+			return errors.Wrap(dbErr, "unable to update the snapshot on report generation failure")
+		}
+		return errors.Wrapf(err, "unable to zip the compliance reports for scan config %s", req.ScanConfigName)
+	}
+	snapshot.GetReportStatus().RunState = storage.ComplianceOperatorReportStatus_GENERATED
+	if err := rg.snapshotDS.UpsertSnapshot(req.Ctx, snapshot); err != nil {
+		return errors.Wrap(err, "unable to update snapshot on report generation success")
```

Comment:
IMO the best thing to do in these scenarios would be to retry. If upserting in the db fails it doesn't mean that the report failed, so I'd argue we should maybe retry and log the error. Also if upserting here, it could also fail upserting the FAILURE state.